src/main/java/com/impactupgrade/nucleus/service/segment/MailchimpCommunicationService.java

```diff
@@ -179,16 +169,60 @@ protected void syncContacts(PagedResults.ResultSet<CrmContact> resultSet, Enviro
     }
   }
 
-  protected void massArchive(EnvironmentConfig.Mailchimp mailchimpConfig,
-      EnvironmentConfig.CommunicationList communicationList, List<MemberInfo> listMembers, Set<String> seenEmails,
-      MailchimpClient mailchimpClient) throws Exception {
-    // get all mc email addresses in the entire audience
-    Set<String> emailsToArchive = listMembers.stream().map(memberInfo -> memberInfo.email_address.toLowerCase(Locale.ROOT)).collect(Collectors.toSet());
-    // remove all CRM emails from the list of MC emails, which leaves us with the list that needs to be archived
-    emailsToArchive.removeAll(seenEmails);
+  @Override
+  public void massArchive() throws Exception {
```

Comment:
@VSydor Once we merge your new pagination processor, here's another two spots where it could be used!